### ISSUE TITLE
[rulename] prefer-const false positive

### ISSUE THREAD

**Issue by Weldify** [typescript-eslint/typescript-eslint#4572]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-const": "error"
  }
}
```

```TS
class Class {}
class ChildClass extends Class {}

namespace Space {
	export let thing: typeof Class = Class;
    // ^ 'thing' is never reassigned. Use 'const' instead. eslint(prefer-const)
}

Space.thing = ChildClass;
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `vscode ESLint` | `2.2.2` |
| `vsocde`        | `1.64.2` |
| `ESLint`                           | `X8.9.0` |

**Comment by bradzacher** [comment]
The rule cannot detect such usages across boundaries.

[Just like the rule cannot and does not detect usages across module boundaries](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IFwicHJlZmVyLWNvbnN0XCI6IFtcImVycm9yXCJdICovXG5cbmV4cG9ydCBsZXQgeCA9IDE7XG5cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWVsc2UtaWYiOjIsIm5vLWR1cGUta2V5cyI6Miwibm8tZHVwbGljYXRlLWNhc2UiOjIsIm5vLWVtcHR5IjoyLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLWVtcHR5LXBhdHRlcm4iOjIsIm5vLWV4LWFzc2lnbiI6Miwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjoyLCJuby1leHRyYS1zZW1pIjoyLCJuby1mYWxsdGhyb3VnaCI6Miwibm8tZnVuYy1hc3NpZ24iOjIsIm5vLWdsb2JhbC1hc3NpZ24iOjIsIm5vLWltcG9ydC1hc3NpZ24iOjIsIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6Miwibm8taW52YWxpZC1yZWdleHAiOjIsIm5vLWlycmVndWxhci13aGl0ZXNwYWNlIjoyLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6Miwibm8tbWl4ZWQtc3BhY2VzLWFuZC10YWJzIjoyLCJuby1uZXctc3ltYm9sIjoyLCJuby1vYmotY2FsbHMiOjIsIm5vLW9jdGFsIjoyLCJuby1wcm90b3R5cGUtYnVpbHRpbnMiOjIsIm5vLXJlZGVjbGFyZSI6Miwibm8tcmVnZXgtc3BhY2VzIjoyLCJuby1zZWxmLWFzc2lnbiI6Miwibm8tc2V0dGVyLXJldHVybiI6Miwibm8tc2hhZG93LXJlc3RyaWN0ZWQtbmFtZXMiOjIsIm5vLXNwYXJzZS1hcnJheXMiOjIsIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjoyLCJuby11bmRlZiI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyfSwiZW52Ijp7fX19)

**Comment by Weldify** [comment]
In that case, why is the rule not disabled for `export`?

**Comment by HolgerJeromin** [comment]
I stumbled over the same problem. Having an ignore-next-line is not very elegant.

https://typescript-eslint.io/play/#ts=4.6.2&sourceType=module&code=LYewJgrgNgpgBAMxCOcDeBYAUK1MAeADiAE4AucAdgIbAwDOh1AxvADIgDm62ufcBYuTiwKojpzYwAbjChwAvHADkAURIlS9AOrUSlAJaVO9AJKUk9ZQG5e-PEVIVmISvWev3EqbPlK1Glq6+kYm5pY2dqgAvtixWEggAHQSSeJcPnKKKgBGyMrYiSlcSS5uZN4yWf55IMpAA&rules=N4IgDgTgpgZlEFoDGB7AdgZwC4gFwnghQhAF8g&tsConfig=N4XyA

```ts
module foo  {
    export namespace Log {
        // 'letLogLevel' is never reassigned. Use 'const' instead. ESLint(prefer-const)
        export let letLogLevel = 'ErrorsWarningsInfos'; 
        export const constLogLevel = 'ErrorsWarningsInfos';
    }
}
// in another file:
foo.Log.letLogLevel = 'boo'
// Cannot assign to 'constLogLevel' because it is a read-only property.(2540)
foo.Log.constLogLevel = 'boo'
```
